packages/premint-sdk/src/premint-api.ts

```diff
@@ -0,0 +1,684 @@
+  network: NetworkConfig;
+  chain: Chain;
+  rewardPerToken: bigint;
+  constructor(chain: Chain) {
+    this.rewardPerToken = BigInt("777000000000000");
+    this.chain = chain;
+    const networkConfig = networkConfigByChain[chain.id];
+    if (!networkConfig) {
+      throw new Error(`Not configured for chain ${chain.id}`);
+    }
+    this.network = networkConfig;
+  }
+
+  /**
+   * The premint executor address is deployed to the same address across all chains.
+   * Can be overridden as needed by making a parent class.
+   *
+   * @returns Executor address for premints
+   */
+  getExecutorAddress() {
+    return zoraCreator1155PremintExecutorImplAddress[999];
+  }
+
+  /**
+   * The fixed price minter address is the same across all chains for our current
+   * deployer strategy.
+   * Can be overridden as needed by making a parent class.
+   *
+   * @returns Fixed price sale strategy
+   */
+  getFixedPriceMinterAddress() {
+    return zoraCreatorFixedPriceSaleStrategyAddress[999];
+  }
+
+  /**
+   * A simple fetch() wrapper for HTTP gets.
+   * Can be overridden as needed.
+   *
+   * @param path Path to run HTTP JSON get against
+   * @returns JSON object response
+   * @throws Error when HTTP response fails
+   */
+  async get(path: string) {
+    const response = await fetch(path, { method: "GET" });
+    if (response.status !== 200) {
+      let json;
+      try {
+        json = await response.json();
+      } catch (e: any) {}
+      throw new BadResponse(
+        `Invalid response, status ${response.status}`,
+        response.status,
+        json,
+      );
+    }
+    return await response.json();
+  }
+
+  /**
+   * A simple fetch() wrapper for HTTP post.
+   * Can be overridden as needed.
+   *
+   * @param path Path to run HTTP JSON POST against
+   * @param data Data to POST to the server, converted to JSON
+   * @returns JSON object response
+   * @throws Error when HTTP response fails
+   */
+  async post(path: string, data: any) {
+    const response = await fetch(path, {
+      method: "POST",
+      headers: {
+        "content-type": "application/json",
+        accept: "application/json",
+      },
+      body: JSON.stringify(data),
+    });
+    if (response.status !== 200) {
+      let json;
+      try {
+        json = await response.json();
+      } catch (e: any) {}
+      throw new BadResponse(
+        `Bad response: ${response.status}`,
+        response.status,
+        json,
+      );
+    }
+    return await response.json();
+  }
+
+  /**
+   * Getter for public client that instantiates a publicClient as needed
+   *
+   * @param publicClient Optional viem public client
+   * @returns Existing public client or makes a new one for the given chain as needed.
+   */
+  getPublicClient(publicClient?: PublicClient): PublicClient {
+    if (publicClient) {
+      return publicClient;
+    }
+    return createPublicClient({ chain: this.chain, transport: http() });
+  }
+
+  /**
+   * Update premint
```

Comment:
id add a bit more details about what this is doing (i.e. replaces existing signature with the new one by bumping the version and signing the new version and saving it to the api).